### ISSUE TITLE
Fix S2583 FP: Loop with manually incremented counter

### ISSUE THREAD

**Issue by zsolt-kolbay-sonarsource** [SonarSource/sonar-dotnet#8028]
### Description

S2583 raises a warning for the following code:

```cs
public void Test(int[] items)
{
    var m = 0;
    foreach (var item in items)
    {
        m++;
        if (m % 3 == 0) // Noncompliant - FP
        {
            // do something
        }
    }
}
```

**Note:** the warning is not raised when 2 is used instead of 3 in the modulo operation:
```cs
...
if (m % 2 == 0) // Does not raise
...
```

### Expected behavior

The analyzer should not raise an issue.

### Actual behavior

The analyzer raises an issue: `S2583: Change this condition so that it does not always evaluate to 'False'.`

### Related information

* C#/VB.NET Plugins version: 9.10.0.77988
* Visual Studio version: Visual Studio 2022 (version 17.7)
* MSBuild / dotnet version: .NET 7.0
* Operating System: Windows 10

**Comment by odalet** [comment]
As said in a duplicate of this issue, this is very dangerous (as people quick to trust Sonar analysis will get rid of the code rather than pondering whether this could be a false positive...)

I just updated from v9.15 to v9.16 and the false positive is still there. I know the work is currently in progress, but I think it'd be worthwhile in the meantime to publish an update of `SonarAnalyzer.CSharp` that simply does not raise the issue at all. Enable it again once you're confident the issue is solved please! 

FYI, here is some code of mine that raises the false positive (I guess my loop should be analyzed more than 4 times for the condition to be deemed necessary):

```csharp
var w = size.Width;
var h = size.Height;
var uData = new float[w * h];
var vData = new float[w * h];
for (var y = 0; y < h; y++)
    for (var x = 0; x < w; x++)
    {
        uData[x + y * w] = 2 * y > h ? y - h : y;
        vData[x + y * w] = 2 * x > w ? x - w : x;
    }
```

Raised on both conditions...
